OWML.ModHelper.Menus/ModInputCombinationMenu.cs

```diff
@@ -0,0 +1,205 @@
+            labelPanel.gameObject.SetActive(false);
+
+            var layoutGroup = menu.GetComponentsInChildren<VerticalLayoutGroup>().Single(x => x.name == "Content");
+            Initialize(menu, layoutGroup);
+
+            var saveButton = GetButton("UIElement-SaveAndExit");
+            var addButton = GetButton("UIElement-ResetToDefaultsButton");
+            var cancelButton = GetButton("UIElement-DiscardChangesButton");
+
+            saveButton.OnClick += OnSave;
+            addButton.OnClick += OnAdd;
+            cancelButton.OnClick += OnExit;
+
+            saveButton.SetControllerCommand(InputLibrary.confirm);
+            cancelButton.SetControllerCommand(InputLibrary.cancel);
+            addButton.SetControllerCommand(InputLibrary.setDefaults);
+
+            var localText = addButton.Button.gameObject.GetComponentInChildren<LocalizedText>(true);
+            if (localText != null)
+            {
+                GameObject.Destroy(localText);
+            }
+            var buttonWithHotkey = addButton.Button.gameObject.GetComponentInChildren<ButtonWithHotkeyImageElement>(true);
+            if (buttonWithHotkey != null)
+            {
+                buttonWithHotkey.SetPrompt(new ScreenPrompt(InputLibrary.setDefaults, "Add Alternative"));
+            }
+
+            Title = "Edit Combination";
+
+            GetButton("UIElement-CancelOutOfRebinding").Hide();
+            GetButton("UIElement-KeyRebinder").Hide();
+
+            for (int i = 0; i < layoutGroup.transform.childCount; i++)
+            {
+                layoutGroup.transform.GetChild(i).gameObject.SetActive(false);
+            }
+        }
+
+        public void RemoveCombinationElement(IModInputCombinationElement element)
+        {
+            CombinationElements.Remove(element);
+            var selectable = element.Toggle.GetComponent<Selectable>();
+            for (int i = 0; i < _selectables.Count; i++)
+            {
+                if (selectable != _selectables[i])
+                {
+                    continue;
+                }
+                var upIndex = (i - 1 + _selectables.Count) % _selectables.Count;
+                var downIndex = (i + 1) % _selectables.Count;
+                var navigation = _selectables[upIndex].navigation;
+                navigation.selectOnDown = _selectables[downIndex];
+                _selectables[upIndex].navigation = navigation;
+                navigation = _selectables[downIndex].navigation;
+                navigation.selectOnUp = _selectables[upIndex];
+                _selectables[downIndex].navigation = navigation;
+                if (downIndex == 0)
+                {
+                    _selectables[upIndex].Select();
+                }
+                else
+                {
+                    _selectables[downIndex].Select();
+                }
+                _selectables.RemoveAt(i);
+                break;
```

Comment:
Extract to new method UpdateNavigation?

Reply:
It's a bit faster that way though

Reply:
wdym?

Reply:
I mean that I don't fully update all navigations, jsut the ones surrounding removed/added one

Reply:
Ah, still sounds like it can be extracted to a new method.